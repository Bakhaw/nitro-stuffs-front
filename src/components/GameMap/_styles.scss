.GameMap {
  display: flex;
  flex: 8;
  flex-direction: column;
  &__Title {
    display: flex;
    margin-bottom: 45px;
    color: $White;
  }
  &__Content {
    display: flex;
    flex: 1;
    flex-direction: row;
    flex-wrap: wrap;

    &__Map {
      display: flex;
      flex: 1;
      &__Shadow {
        opacity: 0.3;
      }
    }
    &__Video {
      display: flex;
      flex: 1;
      align-items: center;
      &__Paragraphe {
        display: flex;
        flex: 1;
        color: white;
        justify-content: center;
      }
    }
  }
}
