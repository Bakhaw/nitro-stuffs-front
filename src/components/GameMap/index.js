import React, { useEffect, useState } from 'react';
import PropTypes from 'prop-types';

import mirage from '../../static/images/csgo/maps-callouts/mirage.jpg';
import MarkerList from '../MarkerList';
import StuffCard from '../StuffCard';

function GameMap({ activeStuffs }) {
  //All Stuff to show with filter
  const [stuffShown, setStuffShown] = useState([]);
  //Stuff selected to show all videos associated
  const [stuffSelected, setStuffSelected] = useState([]);
  //Stuff selected type (smoke,flash, ...)
  const [stuffSelectedType, setStuffSelectedType] = useState('');
  //Boolean to change when user click or leave the selected stuff
  const [isSelectedStuff, setIsSelectedStuff] = useState(true);
  //All videos associated to the stuff selected
  const [allVideos, setAllVideos] = useState([]);
  //The video selected
  const [videoSelected, setVideoSelected] = useState(null);

  useEffect(() => {
    showStuff();
  }, [stuffShown, stuffSelected, activeStuffs]);

  //Stuff to show on the screen: the selected one (stuffSelected) or the default one (activeStuffs: with the filters)
  function showStuff() {
    if (stuffSelected.length) {
      setStuffShown(stuffSelected);
      setIsSelectedStuff(true);
    } else {
      setStuffShown(activeStuffs);
      setIsSelectedStuff(false);
    }
  }

  //Handle click function : Called when a stuff is selected
  //Store the stuff selected marker (calling the useEffect showStuff()), and the markers for all the videos
  function handleClickStuffSelected(stuff) {
    setStuffSelected([stuff]);
    setStuffSelectedType(stuff.type);
    setAllVideos(stuff.stuffPositions);
  }

  //Handle click function : Called when a video marker is selected
  //Store the video selected to show
  function handleClickVideoSelected(videoSelecteds) {
    setVideoSelected(videoSelecteds);
  }

  //Handle click function : Called when you want to remove the selections made
  //Reset all selections made and show the default stuffs (activeStuffs: with the filters)
  function handleClickShowAllStuff() {
    setStuffSelected([]);
    setStuffSelectedType('');
    setAllVideos([]);
    setVideoSelected(null);
  }

  return (
    <div className='GameMap'>
      <div className='GameMap__Title'>
        <h1>MIRAGE</h1>
      </div>
      <div className='GameMap__Content'>
        <div className='GameMap__Content__Map' style={{ position: 'relative' }}>
          <img
            className={`${
              isSelectedStuff ? 'GameMap__Content__Map__Shadow' : ''
            }`}
            alt='Mirage'
            src={mirage}
          />
          <MarkerList
            allVideos={allVideos}
            isSelectedStuff={isSelectedStuff}
            stuffShown={stuffShown}
            handleClickShowAllStuff={handleClickShowAllStuff}
            handleClickStuffSelected={handleClickStuffSelected}
            handleClickVideoSelected={handleClickVideoSelected}
          />
        </div>
        <div className='GameMap__Content__Video'>
          {videoSelected ? (
            <StuffCard stuff={videoSelected} type={stuffSelectedType} />
          ) : (
            <p className='GameMap__Content__Video__Paragraphe'>
              Select a video on the map
            </p>
          )}
        </div>
      </div>
    </div>
  );
}

GameMap.propTypes = {
  activeStuffs: PropTypes.array.isRequired,
};

export default GameMap;
