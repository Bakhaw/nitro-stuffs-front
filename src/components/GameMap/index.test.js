import React from 'react';
import { shallow } from 'enzyme';

import GameMap from '.';

test('should render properly', () => {
  const props = {
    activeStuffs: [],
  };

  const container = shallow(<GameMap {...props} />);
  expect(container.debug()).toMatchSnapshot();
});
