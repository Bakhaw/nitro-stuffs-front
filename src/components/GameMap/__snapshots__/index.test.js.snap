// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render properly 1`] = `
"<div className=\\"GameMap\\">
  <div className=\\"GameMap__Title\\">
    <h1>
      MIRAGE
    </h1>
  </div>
  <div className=\\"GameMap__Content\\">
    <div className=\\"GameMap__Content__Map\\" style={{...}}>
      <img className=\\"GameMap__Content__Map__Shadow\\" alt=\\"Mirage\\" src=\\"mirage.jpg\\" />
      <MarkerList allVideos={{...}} isSelectedStuff={true} stuffShown={{...}} handleClickShowAllStuff={[Function: handleClickShowAllStuff]} handleClickStuffSelected={[Function: handleClickStuffSelected]} handleClickVideoSelected={[Function: handleClickVideoSelected]} />
    </div>
    <div className=\\"GameMap__Content__Video\\">
      <p className=\\"GameMap__Content__Video__Paragraphe\\">
        Select a video on the map
      </p>
    </div>
  </div>
</div>"
`;
